src/lib.rs

```diff
@@ -415,6 +420,54 @@ mod _64 {
     }
 }
 
+#[cfg(feature = "x128")]
+mod _x128 {
+    use {Error, From};
+
+    // Signed
+    promotion! {
+        i8    =>                              i128;
+        i16   =>                              i128;
+        i32   =>                              i128;
+        i64   =>                              i128;
+        isize =>                              i128;
+        i128  => f32, f64,                    i128;
+    }
+
+    half_promotion! {
+        i8    =>                                                              u128;
+        i16   =>                                                              u128;
+        i32   =>                                                              u128;
+        i64   =>                                                              u128;
+        isize =>                                                              u128;
+        i128  =>                                                              u128;
+    }
+
+    from_signed! {
+        i128  =>           i8, i16, i32, i64,       isize, u8, u16, u32, u64,       usize;
+    }
+
+    // Unsigned
+    promotion! {
+        u8    =>                              i128,                           u128;
+        u16   =>                              i128,                           u128;
+        u32   =>                              i128,                           u128;
+        u64   =>                              i128,                           u128;
+        usize =>                              i128,                           u128;
+        u128  => f32, f64,                                                    u128;
```

Comment:
I'm not sure if u128 -> f32 can be really be considered a promotion.

`f32::MAX` prints `340282350000000000000000000000000000000`.
`u128::MAX` prints `340282366920938463463374607431768211455`

So `u128::MAX` is larger than `f32::MAX` or IOW `u128::MAX` doesn't fit in `f32`. IMO, this conversion should return Error::Overflow. However, rustc (and gcc) seem to be fine with the `u128::MAX as f32` operation; it returns `inf`. That seems wrong to me -- that's not loss of precision; that's a rounding operation with a rounding error of `inf`.